elasticapm/instrumentation/packages/dbapi2.py

```diff
@@ -170,30 +170,44 @@ def extract_signature(sql):
         keyword = "INTO" if sql_type == "INSERT" else "FROM"
         sql_type = sql_type + " " + keyword
 
-        table_name = look_for_table(sql, keyword)
+        object_name = look_for_table(sql, keyword)
     elif sql_type in ["CREATE", "DROP"]:
         # 2nd word is part of SQL type
         sql_type = sql_type + sql[first_space:second_space]
-        table_name = ""
+        object_name = ""
     elif sql_type == "UPDATE":
-        table_name = look_for_table(sql, "UPDATE")
+        object_name = look_for_table(sql, "UPDATE")
     elif sql_type == "SELECT":
         # Name is first table
         try:
             sql_type = "SELECT FROM"
-            table_name = look_for_table(sql, "FROM")
+            object_name = look_for_table(sql, "FROM")
         except Exception:
-            table_name = ""
+            object_name = ""
+    elif sql_type in ["EXEC", "EXECUTE"]:
+        sql_type = "EXECUTE"
+        end = second_space if second_space > first_space else len(sql)
+        object_name = sql[first_space + 1 : end]
+    elif sql_type == "CALL":
+        first_paren = sql.find("(", first_space)
```

Comment:
I'm not certain this line is quite right. It assumes that if there is no parenthesis, we'll grab from the first space to the second to last character (as `first_paren` will be `-1`). What if there's no trailing semicolon? Maybe add a test case for `("CALL get_car_stats_by_year", "CALL get_car_stats_by_year()"),`?

Reply:
Good catch! I added the test case and refactored the mentioned part.

Note: In the current commit I have `object_name = procedure_name + "()"`. The added parens are there to be consistent with the already existing code in `_trace_sql` for calls coming from `def callproc`.
If `object_name = procedure_name` is desired, the pre-existing code in _trace_sql should be updated accordingly.

Reply:
I think your version is fine. Thanks!